Yosemite/Yosemite/Stores/AppSettingsStore.swift

```diff
@@ -108,11 +113,67 @@ public class AppSettingsStore: Store {
             setProductsFeatureSwitch(isEnabled: isEnabled, onCompletion: onCompletion)
         case .resetStatsVersionStates:
             resetStatsVersionStates()
+        case .setInstallationDateIfNecessary(let date, let onCompletion):
+            setInstallationDateIfNecessary(date: date, onCompletion: onCompletion)
+        case .setLastFeedbackDate(let date, let onCompletion):
+            setLastFeedbackDate(date: date, onCompletion: onCompletion)
         }
     }
 }
 
+// MARK: - General App Settings
+
+private extension AppSettingsStore {
+    /// Save the `date` in `GeneralAppSettings` but only if the `date` is older than the existing
+    /// `GeneralAppSettings.installationDate`.
+    ///
+    func setInstallationDateIfNecessary(date: Date, onCompletion: ((Result<Void, Error>) -> Void)) {
+        do {
+            let settings = try loadOrCreateGeneralAppSettings()
+
+            let installationDate = settings.installationDate ?? .distantFuture
+            guard date < installationDate else {
+                return onCompletion(.success(()))
```

Comment:
I'm not sure if returning a `success` when the date wasn't expresses the correct intention.

What do you think on returning a `Boolean` to indicated if the date was actually saved or not?

Reply:
We could but I don't expect to use the value. This will only be called on app start. 

I also wanted this function to be just fire and forget. 😅 

I can still add it though! Maybe it can help in unit testing.

Reply:
Added `Bool` argument in 922904ce4528a5b2d9f78563c6136fb7cec628bd.